src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -153,8 +169,9 @@ public static void ReThrowWithPath(in ReadStack readStack, JsonReaderException e
             throw new JsonException(message, path, ex.LineNumber, ex.BytePositionInLine, ex);
         }
 
+        [DoesNotReturn]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void ReThrowWithPath(in ReadStack readStack, in Utf8JsonReader reader, Exception ex)
+        public static void ReThrowWithPath(in ReadStack readStack, in Utf8JsonReader reader, Exception? ex)
```

Comment:
Why does `Exception` need to be nullable?

```suggestion
        public static void ReThrowWithPath(in ReadStack readStack, in Utf8JsonReader reader, Exception ex)
```